mailpoet/tests/integration/Cron/Workers/SchedulerTest.php

```diff
@@ -582,15 +584,19 @@ public function testItCanProcessPostNotificationNewsletter() {
     verify($updatedNotificationHistory->getStatus())->equals(NewsletterEntity::STATUS_SENDING);
   }
 
-  public function testItFailsToProcessWhenScheduledQueuesNotFound() {
-    $scheduler = $this->getScheduler();
-    verify($scheduler->process())->false();
-  }
-
   public function testItDeletesQueueDuringProcessingWhenNewsletterNotFound() {
-    $queue = $this->_createQueue(1);
-    $queue->scheduledAt = Carbon::createFromTimestamp(WPFunctions::get()->currentTime('timestamp'));
-    $queue->save();
+    $newsletter = $this->_createNewsletter();
+    $task = $this->createTaskWithQueue($newsletter);
+    $task->setScheduledAt(Carbon::createFromTimestamp(WPFunctions::get()->currentTime('timestamp')));
+    $this->entityManager->flush();
+
+    // remove newsletter, but not any related data
+    $this->entityManager->getConnection()->delete(
+      $this->entityManager->getClassMetadata(NewsletterEntity::class)->getTableName(),
+      ['id' => $newsletter->getId()]
+    );
+    $this->entityManager->clear();
```

Comment:
Oh, good catch! It was a bug, this test (and the two others) passed even without the delete! It was because there was neither a scheduled task subscriber, not a matching segment, so it deleted the queue in any way. I fixed all of them, and now the deletes and `->clear()` calls are needed.

Fix: https://github.com/mailpoet/mailpoet/pull/5241/commits/fcbd3c81023885952dc189966779335441702a89